File: WebApplication1/Content/components/Crosswords/Models/CrosswordIconViewModel.ts
import {CrosswordModel} from "./CrosswordModel";
import {Solve} from "../../Solve/Models/Solve";
import {LeaderboardItemViewModel} from "./LeaderboardItemViewModel";

export class CrosswordIconViewModel {
    id: string;
    isAnonymous: boolean;
    userId: string;
    title: string;
    author: string;
    grid: string[][];
    solves: LeaderboardItemViewModel[];

    constructor() {
        this.grid = [];
        this.id = "";
        this.isAnonymous = false;
        this.userId = "";
        this.title = "";
        this.author = "";
        this.solves = [];
    }

    constructor(crosswordModel: CrosswordModel) {
        this.grid = crosswordModel.grid.map(row => row.map(cell => cell === '.' ? '.' : ''));
        this.author = crosswordModel.author;
        this.id = crosswordModel.id;
        this.isAnonymous = crosswordModel.isAnonymous;
        this.title = crosswordModel.title;
        this.userId = crosswordModel.userId;
        this.solves = [];
    }
}

File: WebApplication1/ViewModels/Crosswords/CrosswordIconViewModel.cs
using CrossWorldApp.Models;

namespace CrossWorldApp.ViewModels.Crosswords;

public class CrosswordIconViewModel
{
    public int Id { get; set; }
    
    public string UserId { get; set; }
    public string Title { get; set; }
    public string Author { get; set; }
    public bool IsAnonymous { get; set; }
    public List<LeaderboardItemViewModel> Solves { get; set; }
    public List<List<string>> Grid { get; set; }

    public CrosswordIconViewModel(TestCrossword crossword)
    {
        Id = crossword.Id;
        Title = crossword.Title;
        UserId = crossword.UserId;
        Author = crossword.Author;
        IsAnonymous = crossword.IsAnonymous;
        Grid = crossword.Grid;
        Solves = crossword.Solves.Select(solve => new LeaderboardItemViewModel(solve)).ToList();
    }
}